### PR TITLE
fix: [io]Memory leak due to failed asynchronous query for file information

### PR DESCRIPTION
**Pull request by liyigang1**
Memory leak due to failed asynchronous query for file information

Log: Memory leak due to failed asynchronous query for file information

### CODE REVIEW

File: src/dfm-base/utils/fileinfohelper.cpp [hunk 1]
```diff
@@ -99,8 +99,16 @@ void FileInfoHelper::fileRefreshAsyncCallBack(bool success, void *userData)
     if (!userData)
         return;
     auto data = static_cast<FileRefreshCallBackData *>(userData);
-    if (!data->info)
+    if (!success) {
+        qWarning() << "Failed to query file information asynchronously! \n" << (data->info ? "" : data->info->fileUrl().toString());
+        delete data;
```
Comment (Johnson-zs) [inline src/dfm-base/utils/fileinfohelper.cpp]: 这个内存需要自己维护么？此时data是否一定安全非野指针且不为空？建立使用FinallyUtil来封装

Comment (liyigang1) [reply src/dfm-base/utils/fileinfohelper.cpp]: 这里是自己new了再传入的，data是不会为野指针的，前面已做过判空处理了

**Review by Kakueeen — APPROVED** [review]

File: src/dfm-base/utils/fileinfohelper.cpp (hunk 1 above)
Comment (max-lvs) [reply src/dfm-base/utils/fileinfohelper.cpp]: 如果 data 为空呢，日志会崩溃吧

**Review by Johnson-zs — DISMISSED** [review]

**Review by Johnson-zs — APPROVED** [review]

**Pull request merged**